src/libsystemd-network/sd-dhcp-server.c

```diff
@@ -773,8 +777,11 @@ int dhcp_server_handle_message(sd_dhcp_server *server, DHCPMessage *message,
                         return 0;
 
                 /* for now pick a random free address from the pool */
-                if (existing_lease)
+                if(static_lease){
+                        address = static_lease->address;
+                }else if (existing_lease){
                         address = existing_lease->address;
+                }
```

Comment:
coding style off. Please consult https://systemd.io/CODING_STYLE

no {} around single line if blocks. Space after "if" and before "else"...